### PR TITLE
Perform uploads per file, and delay request.

### PR DESCRIPTION
**Pull request by Zmetser**
1. Adds the ability to upload multiple files but perform a single upload request for each of them.
2. Adds the ability to delay a file upload until an other operation finishes.

Both are required to operate this file upload plugin with 3rd party services like cloudinary, where you have to get a token for every file and upload individual files using their secure token.

**singleFileUploads** upload every selected file in it's own request.
**beforeRequest** delay uploading.

Example `beforeRequest` callback:
```javascript
// beforeRequest is called for every file when singleFileUploads is true
function beforeRequest(fileList, formData) {
	// Get token from Cloudinary for the file
	return cloudinaryRequest.getSignedRequest(fileList[0].name).then(function (signedRequest) {
		// Attach token to formData 
		formData.append('token', signedRequest.token);
		// Return with formData to start file upload
		return formData;
	});
}
```

### CODE REVIEW

File: jquery.liteuploader.js [hunk 1]
```diff
@@ -22,7 +28,7 @@ function LiteUploader (element, options) {
     this.el = $(element);
     this.options = options;
     this.params = options.params;
-    this.xhr = this._buildXhrObject();
+    this.xhrs = [];
```
Comment (Zmetser) [inline jquery.liteuploader.js]: This holds all the concurrent XHR Objects, to be able to abort them all by the client, there's no other purpose for this.
The first object is created on the first upload, not before.

File: jquery.liteuploader.js [hunk 2]
```diff
@@ -154,16 +176,13 @@ LiteUploader.prototype = {
...
     },
 
-    _getXHRObject: function () {
-        return this.xhr;
-    },
-
     _performUpload: function (formData) {
         $.ajax({
-            xhr: this._getXHRObject.bind(this),
+            xhr: this._buildXhrObject.bind(this),
```
Comment (Zmetser) [inline jquery.liteuploader.js]: Every upload creates it's own `XMLHttpRequest` instance on the fly. The build function adds the instance to the `xhrs` container.

File: readme.md [hunk 3]
```diff
@@ -74,6 +85,11 @@ The aim was to see, at its absolute minimum, how little code do you need to make
         <td>triggered after input and file validation - see 'File Error Types' section for more</td>
     </tr>
     <tr>
+        <td>lu:start</td>
```
Comment (Zmetser) [inline readme.md]: This event will always be called once per file selection, whereas the `lu:before` might get called before each file upload starts, depending on the `singleFileUploads` flag.

File: tests/spec/jquery.liteuploader.spec.js [hunk 4]
```diff
@@ -81,13 +79,29 @@ describe('Lite Uploader', function () {
             expect(liteUploader._performUpload).not.toHaveBeenCalled();
         });
 
+        it('should not proceed with upload if beforeRequest was rejected', function () {
```
Comment (Zmetser) [inline tests/spec/jquery.liteuploader.spec.js]: New test to make sure `beforeRequest` stops the upload, if the client operation fails.
The success or resolved state is covered in the 'should proceed with upload if no errors are found' test.

File: examples/multiple-single-request.html [hunk 5]
```diff
@@ -0,0 +1,80 @@
+<!DOCTYPE html>
```
Comment (Zmetser) [inline examples/multiple-single-request.html]: An example how to use `singleFileUploads` and `beforeRequest`

**Comment by burt202** [comment]
Yes I like the look of this. I'll try and look at it proper sometime this week but regardless it'll probably get merged. Cheers

**Comment by Zmetser** [comment]
Thanks!

**Pull request merged**